sopel/irc/__init__.py

```diff
@@ -179,29 +179,33 @@ def run(self, host, port=6667):
         try:
             self.backend.run_forever()
         except KeyboardInterrupt:
+            # raised only when the bot is not connected
             LOGGER.warning('Keyboard Interrupt')
-            self.quit('KeyboardInterrupt')
-
-    # Connection Events
+            raise
 
     def on_connect(self):
         """Handle successful establishment of IRC connection."""
+        LOGGER.info('Connected, initiating setup sequence')
+
         # Request list of server capabilities. IRCv3 servers will respond with
         # CAP * LS (which we handle in coretasks). v2 servers will respond with
         # 421 Unknown command, which we'll ignore
+        LOGGER.debug('Sending CAP request')
         self.backend.send_command('CAP', 'LS', '302')
 
         # authenticate account if needed
         if self.settings.core.auth_method == 'server':
+            LOGGER.debug('Sending serveur auth')
             self.backend.send_pass(self.settings.core.auth_password)
         elif self.settings.core.server_auth_method == 'server':
+            LOGGER.debug('Sending serveur auth')
             self.backend.send_pass(self.settings.core.server_auth_password)
 
+        LOGGER.debug('Sending nick "%s"', self.nick)
         self.backend.send_nick(self.nick)
+        LOGGER.debug('Sending user "%s" (name: "%s")', self.user, self.name)
         self.backend.send_user(self.user, '+iw', self.nick, self.name)
 
-        LOGGER.info('Connected.')
```

Comment:
I love the extra logger calls everywhere! Not 100% clear on what removing this one does, though. Do we still have a definitive log line that says the bot is done connecting without this?

Reply:
Yeah, but this was an honest lie: the bot is already connected at the very start of the method, it the setup sequence is just started at the *end* of it: the server need to reply to us about the nick, we still have to join channels, and other stuffs.

So I'd argue that the new logs (I added an INFO log to talk about the init sequence once the bot is connected) is good for *that method*, and that we absolutely need to add way more logs where they belongs (i.e. when we received JOIN acknowledgement for instance).

Reply:
Works for me. Guess it's just the rebase/squash left then.